lib/internal/test_runner/harness.js

```diff
@@ -140,7 +140,12 @@ function setup(root) {
   const rejectionHandler =
     createProcessEventHandler('unhandledRejection', root);
   const coverage = configureCoverage(root, globalOptions);
-  const exitHandler = () => {
+  const exitHandler = async () => {
+    try {
```

Comment:
Two things:

1. It feels awkward to have this be the only hook that is run from this file. I understand why you did that, but I'm almost wondering if we should put this check in the root test's `postRun()` (that function already has root specific code).
2. Should we be ignoring the error here? In the main code path for handling hooks, we only ignore the error after something has failed.

Reply:
1. that will require changing `postRun` to be asynchronous. not sure how easy that is without breaking anything, but I will give it a try
2. I guess that is related to the first question. once the `after` hook is not emitted as part of the `run` function - we are outside of the core test succeed/fail mechanism. I will try tackling that as well

Reply:
I would leave 1 alone for now then. Item 2 seems more like a correctness issue though.